### PR TITLE
LPS-105162 SAML stranger policy impl

### PR DESCRIPTION
**Pull request by stian-sigvartsen**
Hi Carlos. The approach is inspired by the approach taken for OpenSSO because both of these avoid rendering the login portlet. There was an opportunity to improve on OpenSSO's approach in 714ff6a . OpenSSO would do a logout action there, whereas with SAML we can request "force Authn". With a Liferay IdP the handling of that results in a logout (area of future improvement), but probably not with most IdP services.

For the design / details on the new unknownUsersAreStrangers IdP connection property agreed with Zsigmond, please see 
https://docs.google.com/document/d/1gCSPnuWs4LsbptkTMPl6Zb6NqTnb_76sR5wryQNlxCM/edit

### CODE REVIEW

**Comment by stian-sigvartsen** [comment]
ci:test:relevant

**Comment by csierra** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/dxp/apps/saml/saml-api/src/main/java/com/liferay/saml/constants/SamlWebKeys.java [hunk 1]
```diff
@@ -77,6 +79,9 @@
 
 	public static final String SAML_SSO_SESSION_ID = "SAML_SSO_SESSION_ID";
 
+	public static final String SAML_SUBJECT_SCREEN_NAME =
+		"SAML_SUBJECT_SCREEN_NAME";
```
Comment (stian-sigvartsen) [inline modules/dxp/apps/saml/saml-api/src/main/java/com/liferay/saml/constants/SamlWebKeys.java]: @csierra  I just remembered that I actually intended to rename this attribute to SAML_SUBJECT_NAME_ID to be more aligned with the SAML concept. Just for consideration.

**Comment by csierra** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by csierra** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/83538. [See changes here.](https://github.com/csierra/liferay-portal/compare/28be1da7d710~2...28be1da7d710) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by csierra** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/83543. [See changes here.](https://github.com/csierra/liferay-portal/compare/392702fb465b~2...392702fb465b) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Pull request closed without merging**